specs/interop.md

```diff
@@ -0,0 +1,540 @@
+      - [Initiating Messages](#initiating-messages)
+    - [CrossL2Inbox](#crossl2inbox)
+      - [Invariants](#invariants)
+        - [Only EOA Invariant](#only-eoa-invariant)
+      - [Executing Messages](#executing-messages)
+    - [L2ToL2CrossDomainMessenger](#l2tol2crossdomainmessenger)
+      - [Transferring Ether in a Cross Chain Message](#transferring-ether-in-a-cross-chain-message)
+      - [Interfaces](#interfaces)
+        - [Sending Messages](#sending-messages)
+        - [Executing Messages](#executing-messages-1)
+    - [L1Block](#l1block)
+  - [SystemConfig](#systemconfig)
+    - [`DEPENDENCY_SET` UpdateType](#dependency_set-updatetype)
+  - [L1Attributes](#l1attributes)
+  - [Fault Proof](#fault-proof)
+  - [Sequencer Policy](#sequencer-policy)
+  - [Block Building](#block-building)
+  - [Sponsorship](#sponsorship)
+  - [Security Considerations](#security-considerations)
+    - [Forced Inclusion of Cross Chain Messages](#forced-inclusion-of-cross-chain-messages)
+    - [Cross Chain Message Latency](#cross-chain-message-latency)
+    - [Dynamic Size of L1 Attributes Transaction](#dynamic-size-of-l1-attributes-transaction)
+    - [Maximum Size of the Dependency Set](#maximum-size-of-the-dependency-set)
+    - [Mempool](#mempool)
+    - [Reliance on History](#reliance-on-history)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+# Interoperability
+
+The ability for a blockchain to easily read the state of another blockchain is called interoperability.
+Low latency interoperability allows for horizontally scalable blockchains, a key feature of the superchain.
+
+Note: this document references an "interop network upgrade" as a temporary name. The actual name of the
+network upgrade will be included in this document in the future.
+
+## Overview
+
+| Term          | Definition                                               |
+|---------------|----------------------------------------------------------|
+| Source Chain   | A blockchain that includes an initiating message |
+| Destination Chain  | A blockchain that includes an executing message |
+| Initiating Message | A transaction submitted to a source chain that authorizes execution on a destination chain |
+| Executing Message | A transaction submitted to a destination chain that corresponds to an initiating message |
+| Cross Chain Message | The cumulative execution and side effects of the initiating message and executing message |
+| Dependency Set | The set of chains that originate initiating transactions where the executing transactions are valid |
+
+A total of two transactions are required to complete a cross chain message. The first transaction is submitted
+to the source chain which authorizes execution on a destination chain. The second transaction is submitted to a
+destination chain, where the block builder SHOULD only include it if they are certain that the first transaction was
+included in the source chain.
```

Comment:
Note: this current design doesn't guarantee atomicity (eg the first transaction doesn't _guarantee_ the second transaction's success, in the same way that an L2 withdrawal doesn't guarantee L1 finalization). 

Atomicity is a potential project on top of this minimal protocol.